modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/ExpressionFactoryImpl.java

```diff
@@ -277,8 +277,17 @@ public Supplier<RowT> rowSource(List<RexNode> values) {
     /** {@inheritDoc} */
     @Override
     public <T> Supplier<T> execute(RexNode node) {
-        RelDataType nodeType = node.getType();
-        RowSchema rowSchema = TypeUtils.rowSchemaFromRelTypes(List.of(nodeType));
+        RelDataType exprType = node.getType();
+        List<RelDataType> typesList;
+
+        if (exprType.getSqlTypeName() == SqlTypeName.ROW) {
```

Comment:
we have no such a check previously, can you explain why do we need it for now ?

Reply:
on the next line after this line there is a comment explaining why this condition is needed.

Previously, RowSchema was not used at all.
Currently it used to resolve object types in the row.
We faced with the problem that table function may return composite type.
As far as I understand, we do not and do not plan to support such types, and at the moment this composite type can only include one "inner" type. Therefore, we decided to convert such a composite type to single base type here.